mono/utils/mono-threads-state-machine.c

```diff
@@ -192,13 +185,13 @@ mono_threads_transition_attach (MonoThreadInfo* info)
 	UNWRAP_THREAD_STATE (raw_state, cur_state, suspend_count, no_safepoints, info);
 	switch (cur_state) {
 	case STATE_STARTING:
-		if (!(suspend_count == 0))
+		if (suspend_count != 0)
 			mono_fatal_with_history ("suspend_count = %d, but should be == 0", suspend_count);
 		if (no_safepoints)
 			mono_fatal_with_history ("no_safepoints = TRUE, but should be FALSE");
 		if (thread_state_cas (&info->thread_state, build_thread_state (STATE_RUNNING, 0, 0), raw_state) != raw_state)
 			goto retry_state_change;
-		trace_state_change ("ATTACH", info, raw_state, STATE_RUNNING, FALSE, 0);
+		trace_state_change_with_func ("ATTACH", info, raw_state, STATE_RUNNING, FALSE, 0, "");
```

Comment:
One other note.  While this PR addresses the literal FIXME to replace uses of `trace_state_change` by `trace_stante_change_with_func`, a better fix would be:
1.  add a `const char *func` argument to these functions like `mono_threads_transition_attach`  (so:  `void mono_threads_transition_attach (MonoThreadInfo* info, const char *func)`),
2. and call `trace_state_change_with_func(..., func);` 
3. and then update every **caller** from `mono_threads_transition_attach (info)` to `mono_threads_transition_attach (info, __func__)`  so that we get the caller's name in the traces.

And the same for the other places where we call `trace_state_change_with_func (..., "")`.